packages/protocol/contracts/common/FixidityLib.sol

```diff
@@ -294,21 +284,16 @@ library FixidityLib {
   /**
    * @notice x/y. If the dividend is higher than maxFixedDividend() it
    * might overflow. You can use multiply(x,reciprocal(y)) instead.
-   * There is a loss of precision on division for the lower mulPrecision() decimals.
    * @dev
-   * NOTE: reverts if the dividend is greater than maxFixedDivisor (10**24,
-   * internally represented with 10**48). 10**6 * 10**18 (so a million units of
-   * a token in wei) is already too large.
    * Test divide(fixed1(),0) fails
    * Test divide(maxFixedDividend(),1) = maxFixedDividend()*(10^digits())
    * Test divide(maxFixedDividend()+1,1) throws
-   * Test divide(maxFixedDivisor(),maxFixedDivisor()) returns fixed1()
-   * Test divide(maxFixedDivisor(),maxFixedDivisor()+1) fails
    */
   function divide(Fraction memory x, Fraction memory y) internal pure returns (Fraction memory) {
-    if (y.value == FIXED1_UINT) return x;
-    require(y.value <= maxFixedDivisor());
-    return multiply(x, reciprocal(y)); // check for 0 done in reciprocal
+    require(y.value != 0);
+    uint256 X = x.value * FIXED1_UINT;
+    require(X / FIXED1_UINT == x.value);
```

Comment:
is this overflow more expensive than checking the max divisor prior?

Reply:
Ah, might be, good catch.